pkg/geo/geomfn/topology_operations.go

```diff
@@ -188,3 +193,75 @@ func SharedPaths(a geo.Geometry, b geo.Geometry) (geo.Geometry, error) {
 	}
 	return gm, nil
 }
+
+// Node Returns a geometry containing a set of linestrings using the least possible number of nodes while preserving all of the input ones.
+func Node(g geo.Geometry) (geo.Geometry, error) {
+	if g.ShapeType() != geopb.ShapeType_LineString && g.ShapeType() != geopb.ShapeType_MultiLineString {
+		return geo.Geometry{}, errors.New("geometry type is unsupported. Please pass a LineString or a MultiLineString")
+	}
+	ep, err := extractUniqueEndpoints(g)
+	if err != nil {
+		return geo.Geometry{}, fmt.Errorf("error extracting unique endpoints: %v", err)
+	}
+
+	res, err := geos.Node(g.EWKB())
+	if err != nil {
+		return geo.Geometry{}, err
+	}
+	node, err := geo.ParseGeometryFromEWKB(res)
+	if err != nil {
+		return geo.Geometry{}, err
+	}
+
+	res, err = geos.LineMerge(node.EWKB())
+	if err != nil {
+		return geo.Geometry{}, err
+	}
+	lines, err := geo.ParseGeometryFromEWKB(res)
+	if err != nil {
+		return geo.Geometry{}, err
+	}
+	if lines.ShapeType() == geopb.ShapeType_LineString {
+		// No nodes found, return a multilinestring
```

Comment:
nit: call this MultiLineString, `.` at end of comment

Reply:
done